src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -1205,6 +1214,52 @@ class NPlusOneDBSpanDetectorExtended(NPlusOneDBSpanDetector):
     )
 
 
+class FileIOMainThreadDetector(PerformanceDetector):
+    """
+    Checks for a file io span on the main thread
+    """
+
+    __slots__ = ("spans_involved", "stored_problems")
+
+    settings_key = DetectorType.FILE_IO_MAIN_THREAD
+
+    def init(self):
+        self.spans_involved = {}
+        self.most_recent_start_time = {}
+        self.most_recent_hash = {}
+        self.stored_problems = {}
+
+    def visit_span(self, span: Span):
+        if self._is_file_io_on_main_thread(span):
+            settings_for_span = self.settings_for_span(span)
+            if not settings_for_span:
+                return
+
+            op, span_id, op_prefix, span_duration, settings = settings_for_span
+            if span_duration.total_seconds() * 1000 > settings["duration_threshold"]:
+                call_stack = ".".join(
+                    [item["function"] for item in span.get("data", {}).get("call_stack", [])]
```

Comment:
I guess we should also consider using `filename` or `module` for the fingerprint, because different classes can have identical function names in theory